utilities/xmlrpcpp/src/XmlRpcValue.cpp

```diff
@@ -610,13 +611,45 @@ namespace XmlRpc {
       default:           break;
       case TypeBoolean:  os << _value.asBool; break;
       case TypeInt:      os << _value.asInt; break;
-      case TypeDouble:   os << _value.asDouble; break;
+      case TypeDouble:
+        {
+          static std::once_flag once;
+          bool use_default = false;
+          char buf[128]; // Should be long enough
+          int required_size = std::snprintf(buf, sizeof(buf)-1,
+                                getDoubleFormat().c_str(), _value.asDouble);
+          if (required_size < 0) {
+            use_default = true;
+            std::call_once(once,
+              [](){XmlRpcUtil::error("Failed to format with %s, using default %%.16g format",
+                getDoubleFormat().c_str());});
```

Comment:
this leads not to detect second or after error format in the same process space. this is unlikely to happen so catching 1st error format would be okay. what would you think?